### PR TITLE
feat(test):add test for client cmd

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description
image

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [x] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: scripts/client_test/cmds/project_cmd.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    def select(self, project: str) -> bool:
+        """
+        :param project:
+        :return:
+            res is:select instance:local, project:self successfully
+                or:failed to select self2, reason: need to create project self2
+        """
+        _valid_str = "successfully"
+        res, err = invoke([CLI, self.project_cmd, "select", project])
+        return True if not err and _valid_str in res else False
```
Comment (tianweidut) [inline scripts/client_test/cmds/project_cmd.py:96]:
```
return True if not err and _valid_str in res else False
```
-->
```
return not err and _valid_str in res
```

Comment (goldenxinxing) [reply scripts/client_test/cmds/project_cmd.py]: done

File: scripts/client_test/cmds/artifacts_cmd.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+        _res, _err = invoke(_args)
+        return json.loads(_res) if not _err else []
+
+    def remove(self, uri: str, force: bool) -> bool:
+        _valid_str = "do successfully"
+        _args = [CLI, self.name, "remove", uri]
+        if force:
+            _args.append("--force")
+        _res, _err = invoke_with_react(_args)
+        return True if not _err and _valid_str in _res else False
```
Comment (tianweidut) [inline scripts/client_test/cmds/artifacts_cmd.py:64]:
```
return not _err and _valid_str
```

File: scripts/client_test/cmds/base/invoke.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import subprocess
+from typing import List, Tuple
+
+
+def invoke(args: List[str]) -> Tuple[str, str]:
+    process = subprocess.run(args,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             universal_newlines=True)
+    # print(f"out:{process.stdout}, error:{process.stderr}")
```
Comment (tianweidut) [inline scripts/client_test/cmds/base/invoke.py:10]: remove this comment?

File: scripts/client_test/cmds/base/invoke.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    return process.stdout, process.stderr
+
+
+def invoke_with_react(args: List[str], input_content: str = "yes") -> Tuple[str, str]:
+    p = subprocess.Popen(args,
+                         stdout=subprocess.PIPE, stdin=subprocess.PIPE,
+                         stderr=subprocess.PIPE, universal_newlines=True)
+    _stdout, _err = p.communicate(input=input_content)
+    # print(f"out:{_stdout}, error:{_err}")
```
Comment (tianweidut) [inline scripts/client_test/cmds/base/invoke.py:19]: remove this comment?

File: scripts/client_test/cmds/eval_cmd.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        if fullname:
+            _args.append("--fullname")
+        if show_removed:
+            _args.append("--show-removed")
+
+        _res, _err = invoke(_args)
+        return json.loads(_res) if not _err else []
+
+    def cancel(self) -> Tuple[str, str]:
+        return invoke([CLI, self._cmd, "cancel", "", "", "", "", "", "", ])
```
Comment (tianweidut) [inline scripts/client_test/cmds/eval_cmd.py:182]: why are there so many empty arguments?

**Comment by tianweidut** [comment]
@goldenxinxing 
1. add python linters for `scripts/client_tests`?
2. how to run the client tests?
3. what are the differences between `scripts/client_tests` and `scripts/run_demo.sh`?
4. should we add `scripts/client_tests` into GitHub actions?

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**